diff --git a/node_modules/react-native-tab-view/lib/typescript/TabBarItem.d.ts b/node_modules/react-native-tab-view/lib/typescript/TabBarItem.d.ts
index f2150eb..878fedd 100644
--- a/node_modules/react-native-tab-view/lib/typescript/TabBarItem.d.ts
+++ b/node_modules/react-native-tab-view/lib/typescript/TabBarItem.d.ts
@@ -29,6 +29,7 @@ export declare type Props<T extends Route> = {
     onLongPress: () => void;
     labelStyle?: StyleProp<TextStyle>;
     style: StyleProp<ViewStyle>;
+    tabContainerStyle?: StyleProp<ViewStyle>;
 };
 export default class TabBarItem<T extends Route> extends React.Component<Props<T>> {
     private getActiveOpacity;
diff --git a/node_modules/react-native-tab-view/src/SceneView.tsx b/node_modules/react-native-tab-view/src/SceneView.tsx
index 4648bf7..d7330b6 100644
--- a/node_modules/react-native-tab-view/src/SceneView.tsx
+++ b/node_modules/react-native-tab-view/src/SceneView.tsx
@@ -137,6 +137,5 @@ export default class SceneView<T extends Route> extends React.Component<
 const styles = StyleSheet.create({
   route: {
     flex: 1,
-    overflow: 'hidden',
   },
 });
diff --git a/node_modules/react-native-tab-view/src/TabBar.tsx b/node_modules/react-native-tab-view/src/TabBar.tsx
index f1137f4..c30704c 100644
--- a/node_modules/react-native-tab-view/src/TabBar.tsx
+++ b/node_modules/react-native-tab-view/src/TabBar.tsx
@@ -304,7 +304,7 @@ export default class TabBar<T extends Route> extends React.Component<
     const { routes } = navigationState;
 
     const isWidthDynamic = this.getFlattenedTabWidth(tabStyle) === 'auto';
-    const tabBarWidth = this.getTabBarWidth(this.props, this.state);
+    const tabBarWidth = this.getTabBarWidth(this.props, this.state) || 1;
     const tabBarWidthPercent = `${routes.length * 40}%`;
     const translateX = this.getTranslateX(
       this.scrollAmount,
diff --git a/node_modules/react-native-tab-view/src/TabBarItem.tsx b/node_modules/react-native-tab-view/src/TabBarItem.tsx
index 80d2e1c..02a6236 100644
--- a/node_modules/react-native-tab-view/src/TabBarItem.tsx
+++ b/node_modules/react-native-tab-view/src/TabBarItem.tsx
@@ -39,6 +39,7 @@ export type Props<T extends Route> = {
   onLongPress: () => void;
   labelStyle?: StyleProp<TextStyle>;
   style: StyleProp<ViewStyle>;
+  tabContainerStyle: StyleProp<ViewStyle>;
 };
 
 const DEFAULT_ACTIVE_COLOR = 'rgba(255, 255, 255, 1)';
@@ -216,7 +217,7 @@ export default class TabBarItem<T extends Route> extends React.Component<
 
     const tabStyle = StyleSheet.flatten(style);
     const isWidthSet = tabStyle?.width !== undefined;
-    const tabContainerStyle: ViewStyle | null = isWidthSet ? null : { flex: 1 };
+    const tabContainerStyle = this.props.tabContainerStyle || (isWidthSet ? null : { flex: 1 });
 
     const scene = { route };
 
@@ -231,7 +232,7 @@ export default class TabBarItem<T extends Route> extends React.Component<
 
     return (
       <PlatformPressable
-        android_ripple={{ borderless: true }}
+        android_ripple={{ borderless: false }}
         testID={getTestID(scene)}
         accessible={getAccessible(scene)}
         accessibilityLabel={accessibilityLabel}
